pkg/roachprod/install/cockroach.go

```diff
@@ -1671,3 +1685,10 @@ func getEnvVars() []string {
 func SuppressMetamorphicConstantsEnvVar() string {
 	return config.DisableMetamorphicTestingEnvVar
 }
+
+// ClusterHook exposes a way to run functions in between steps normally
+// orchestrated by the framework.
+type ClusterHook struct {
```

Comment:
Nit: why not `PrestartHook`? The `Name` is ambiguous, otherwise; plus, the `func` isn't aware of any cluster.

Reply:
I envisioned this being used in `Stop` as well for clean up purposes. Maybe `PreStartStopHook` similar to how we have `StartStopOpts`? I'm okay with just `PrestartHook` for now though until we have other use cases.